### PR TITLE
Traffic Analysis

### PR DESCRIPTION
**Pull request by DanielMwendwa**
[Broadcast Schedule](https://docs.google.com/spreadsheets/d/13ZjhOUdfw0XhreUHQrQsNa2hwDq1D2s9/edit#gid=1769418803)

### CODE REVIEW

File: configurations/pipeline_configuration.py [hunk 1]
```diff
@@ -269,10 +273,100 @@
...
+        traffic_labels=[
+            # S01E01
+            # --------------------------------------- 12 - 13 March -----------------------------------------------                                       
+            TrafficLabel(isoparse("2022-03-12T00:00+03:00"), isoparse("2022-03-12T23:59+03:00"), "E01 Sat, Promo"),
+            TrafficLabel(isoparse("2022-03-13T00:00+03:00"), isoparse("2022-03-13T23:59+03:00"), "E01 Sun, Promo"),
+            # --------------------------------------- 14 March ---------------------------------------------------
+            TrafficLabel(isoparse("2022-03-14T00:00+03:00"), isoparse("2022-03-14T23:59+03:00"), "E01 Mon, None"),
+            # --------------------------------------- 15 March ---------------------------------------------------
+            TrafficLabel(isoparse("2022-03-15T00:00+03:00"), isoparse("2022-03-15T09:19+03:00"), "E01 Tue, None"),
+            TrafficLabel(isoparse("2022-03-15T09:20+03:00"), isoparse("2022-03-15T10:59+03:00"), "E01 Tue, Ad, Group A"),
```
Comment (IsaackMwenda) [inline configurations/pipeline_configuration.py]: Can we add the total of the people we triggered the ad to here? You can pick this from textit flow runs after splitting consent withdrawn. 

```suggestion
            TrafficLabel(isoparse("2022-03-15T09:20+03:00"), isoparse("2022-03-15T10:59+03:00"), "E01 Tue, Ad, Group A- (104xxx)"),
```

File: configurations/pipeline_configuration.py [hunk 2]
```diff
@@ -269,10 +273,100 @@
...
+            TrafficLabel(isoparse("2022-03-29T00:00+03:00"), isoparse("2022-03-29T10:59+03:00"), "E03 Tue, None"),
+            TrafficLabel(isoparse("2022-03-29T11:00+03:00"), isoparse("2022-03-29T23:59+03:00"), "E03 Tue, Radio"),
+            # --------------------------------------- 30 - 01 April ----------------------------------------------
+            TrafficLabel(isoparse("2022-03-30T00:00+03:00"), isoparse("2022-03-30T23:59+03:00"), "E03 Wed, None"),
+            TrafficLabel(isoparse("2022-03-31T00:00+03:00"), isoparse("2022-03-31T23:59+03:00"), "E03 Thur, None"),
+            TrafficLabel(isoparse("2022-04-01T00:00+03:00"), isoparse("2022-04-01T23:59+03:00"), "E03 Fri, None"),
+
+            # S01E04
+            # --------------------------------------- 02 - 03 April ----------------------------------------------- 
+            TrafficLabel(isoparse("2022-04-02T00:00+03:00"), isoparse("2022-04-02T23:59+03:00"), "E04 Sat, Promo"),
```
Comment (IsaackMwenda) [inline configurations/pipeline_configuration.py]: This is useful to check promos' response rate by day - 👍 .

**Review by IsaackMwenda — APPROVED** [review]
lgtm with a few suggestions.

File: configurations/pipeline_configuration.py (hunk 1 above)
Comment (DanielMwendwa) [reply configurations/pipeline_configuration.py]: I have updated the labels.

File: configurations/pipeline_configuration.py [hunk 3]
```diff
@@ -277,10 +281,100 @@
...
-        ws_correct_dataset_code_scheme=load_code_scheme("ws_correct_dataset")
-    ),
-    archive_configuration=ArchiveConfiguration(
-        archive_upload_bucket="gs://pipeline-execution-backup-archive",
-        bucket_dir_path="2022/RVI-ELECTIONS"
+        ws_correct_dataset_code_scheme=load_code_scheme("ws_correct_dataset"),
+        traffic_labels=[
+            # S01E01
+            # --------------------------------------- 12 - 13 March -----------------------------------------------                                       
+            TrafficLabel(isoparse("2022-03-12T00:00+03:00"), isoparse("2022-03-12T23:59+03:00"), "E01 Sat, Promo"),
```
Comment (as2388) [inline configurations/pipeline_configuration.py]: Note that this won't count messages sent after 23:59 on this day. This probably won't change many of your totals, but if you want to capture the whole day it's better to end with 2022-03-12T24:00+03:00

File: configurations/pipeline_configuration.py [hunk 4]
```diff
@@ -277,10 +281,100 @@
...
+            TrafficLabel(isoparse("2022-03-13T00:00+03:00"), isoparse("2022-03-13T23:59+03:00"), "E01 Sun, Promo"),
+            # --------------------------------------- 14 March ---------------------------------------------------
+            TrafficLabel(isoparse("2022-03-14T00:00+03:00"), isoparse("2022-03-14T23:59+03:00"), "E01 Mon, None"),
+            # --------------------------------------- 15 March ---------------------------------------------------
+            TrafficLabel(isoparse("2022-03-15T00:00+03:00"), isoparse("2022-03-15T09:19+03:00"), "E01 Tue, None"),
+            TrafficLabel(isoparse("2022-03-15T09:20+03:00"), isoparse("2022-03-15T10:59+03:00"), "E01 Tue, Ad, Group A (6,286)"),
+            TrafficLabel(isoparse("2022-03-15T11:00+03:00"), isoparse("2022-03-15T23:59+03:00"), "E01 Tue, Radio & Ad"),
```
Comment (as2388) [inline configurations/pipeline_configuration.py]: This misses the minute before the show is scheduled to start, which probably does translate to a few messages not counted

**Review by as2388 — APPROVED** [review]

File: configurations/pipeline_configuration.py (hunk 3 above)
Comment (DanielMwendwa) [reply configurations/pipeline_configuration.py]: Thanks, I've updated the traffic labels.

File: configurations/pipeline_configuration.py (hunk 4 above)
Comment (DanielMwendwa) [reply configurations/pipeline_configuration.py]: Thanks, I've updated this too.

**Pull request merged**